src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1

```diff
@@ -534,6 +535,93 @@ class Project: ADOSVTBase
         return $controlResult
     }
 
+    hidden [ControlResult] CheckMaxBACount([ControlResult] $controlResult){
+        $controlResult.VerificationResult = [verificationResult]::Failed;
+        $TotalBAMembers = 0;
+        if ($this.BAMembers.Count -eq 0) {
+            $this.BAMembers += @([AdministratorHelper]::GetTotalBAMembers($this.OrganizationContext.OrganizationName,$this.ResourceContext.ResourceName))
+        }
+        if($this.BAMembers.Count -gt 0)
+        {
+            $TotalBAMembers = $this.BAMembers.Count
+            $controlResult.AddMessage("There are a total of $TotalBAMembers Build Administrators in your project.")
+            $controlResult.SetStateData("Count of Build Administrators: ",$TotalBAMembers)
+        }
+
+
+        if ($TotalBAMembers -gt 0)
+        {
+            if ([IdentityHelpers]::hasGraphAccess)
+            {
+                $SvcAndHumanAccounts = [IdentityHelpers]::DistinguishHumanAndServiceAccount($this.BAMembers, $this.OrganizationContext.OrganizationName)
+                $humanAccounts = @($SvcAndHumanAccounts.humanAccount | Select-Object displayName, mailAddress)
+                $svcAccounts = @($SvcAndHumanAccounts.serviceAccount | Select-Object displayName, mailAddress)
+
+                $humanAccountsCount = $humanAccounts.Count
+                $svcAccountsCount = $svcAccounts.Count
+                #In case of graph access we will only evaluate the control on the basis of human accounts
+                if($humanAccountsCount -gt $this.ControlSettings.Project.MaxBAMembersPermissible){
+                    $controlResult.AddMessage([VerificationResult]::Failed,"Number of human build administrators configured are more than the approved limit: $($this.ControlSettings.Project.MaxBAMembersPermissible)");
+                }
+                else{
+                    $controlResult.AddMessage([VerificationResult]::Passed,"Number of human build administrators configured are within than the approved limit: $($this.ControlSettings.Project.MaxBAMembersPermissible)");
+                }
+
+                if($TotalBAMembers -gt 0){
```

Comment:
as this condition has already checked if($TotalBAMembers -gt 0) does it needed again?